src/server/engine_shard_set.h

```diff
@@ -144,6 +144,18 @@ class EngineShard {
   void TEST_EnableHeartbeat();
 
  private:
+  struct DefragTaskArgs {
+    uint64_t current_mem_commit = 0u;
+    uint64_t current_mem_used = 0u;
+    uint64_t cursor = 0u;
+
+    void Init();
+
+    // check the current threshold and return true if
+    // we need to do the de-fermentation
```

Comment:
defragmentation, not de-fermentation

Reply:
for some reason the spell check don't like defragmentation